fastlane/lib/fastlane/actions/carthage.rb

```diff
@@ -11,8 +11,13 @@ def self.run(params)
 
         if command_name == "archive" && params[:frameworks].count > 0
           cmd.concat(params[:frameworks])
-        elsif ["update", "build", "bootstrap"].include?(command_name) && params[:dependencies].count > 0
-          cmd.concat(params[:dependencies])
+        elsif ["update", "build", "bootstrap"].include?(command_name)
```

Comment:
Maybe we could add a comment to highlight that `update, build and bootstrap` are the only ones that support `--derived-data` parameter? 

```suggestion
        # "update", "build" and "bootstrap" are the only commands that support "--derived-data" parameter
        elsif ["update", "build", "bootstrap"].include?(command_name)
```

Reply:
Great idea!